packages/common/src/allo/transaction-sender.ts

```diff
@@ -111,10 +118,12 @@ export function createViemTransactionSender(
 ): TransactionSender {
   return {
     async send(tx: TransactionData): Promise<Hex> {
-      const [address] = await walletClient.getAddresses();
+      const account = await walletClient.account;
+
+      if (!account) return "0x";
```

Comment:
Correct me if I am wrong but if the wallet client doesn't have an address this seems like an unexpected error, or is this expected to happen?

If so we should throw an error here, because if we don't the application will fail downstream unexpectedly because the transaction hash is not valid.